components/automate-gateway/gateway/middleware/auth_interceptor.go

```diff
@@ -37,60 +36,6 @@ func NewAuthInterceptor(
 	return &authInterceptor{authn: authn, authz: authz}
 }
 
-type SwitchingAuthorizationHandler interface {
-	GRPCAuthorizationHandler
-	SwitchingFilterHandler
-	IsAuthorized(ctx context.Context, subjects []string,
-		resource, action string, projects []string) (AnnotatedAuthorizationResponse, error)
-}
-
-type SwitchingFilterHandler interface {
-	FilterAuthorizedPairs(ctx context.Context, subjects []string,
-		mapByResourceAndAction map[pairs.Pair][]string,
-		methodsInfo map[string]pairs.Info,
-	) (*FilterPairsResponse, error)
-}
-
-type FilterPairsResponse struct {
-	Pairs                  []*pairs.Pair
-	MethodsInfo            map[string]pairs.Info
-	MapByResourceAndAction map[pairs.Pair][]string
-}
-
-type FilterProjectsResponse struct {
```

Comment:
❓ This type was *unused* -- and there's not much project-iness in the introspection code either. I haven't fully understood the situation... is this in order? @msorens, would you know more about this? What am I missing?

Reply:
https://buildkite.com/chef/chef-automate-master-verify-private/builds/11412#d96352e3-6c8b-4a5f-84fe-3b32cdbfe439/441-1271 👈 So cypress fails on some project-related stuff. I will revisit this, I suppose. At any rate, if someone more familiar with this knows where I took the wrong turn; or if these failures really are "just flakiness", I'd appreciate the insights.

Reply:
The flaky build of cypress always fails. We have flaky and non-flaky sets of the cypress tests. The delta in flaky is either known to sometimes fail or is broken and a WIP. Nothing in there for your build looks out of the ordinary to me.